### PR TITLE
Allow oecert for out-of-proc quoting for SGX

### PR DESCRIPTION
**Pull request by qiucwang**
Add quote-proc param to `oecert` so that `oecert` can specify using in-proc or out-of-proc quoting.

Signed-off-by: Qiucheng Wang <qiucwang@microsoft.com>

### CODE REVIEW

File: tests/tools/oecert/host/host.cpp
Comment (mingweishih) [inline tests/tools/oecert/host/host.cpp:30]: Consider using --out-of-proc instead.

File: tests/tools/oecert/host/host.cpp
Comment (mingweishih) [inline tests/tools/oecert/host/host.cpp:438]: A high-level question: what happens if the out of proc is not supported by the environment? Is there a way to check such cases?

File: tests/tools/oecert/host/host.cpp
Comment (qiucwang) [reply tests/tools/oecert/host/host.cpp:30]: 👍

File: tests/tools/oecert/host/host.cpp
Comment (qiucwang) [reply tests/tools/oecert/host/host.cpp:438]: `out-of-proc` depends on `sgx-quote-ex` library. If it's not installed, it will return sgx-quote-ex not found and quote generation will fail. Ryan's PR #3882 has added those dependencies in ansible script.

Comment (mingweishih) [reply tests/tools/oecert/host/host.cpp:438]: What happens if users simply update the openenclave package without using ansible?

Comment (mingweishih) [reply tests/tools/oecert/host/host.cpp:438]: Ah, I guess sgx-quote-ex should be one of the dependencies then. Is that correct?

Comment (qiucwang) [reply tests/tools/oecert/host/host.cpp:438]: Yes, that's discussed at SIG-Attestation meeting. We will document this behavior so that user will know sgx-quote-ex is the dependencies for out-of-proc quoting.

File: tests/tools/oecert/host/host.cpp
Comment (yentsanglee) [reply tests/tools/oecert/host/host.cpp:30]: We need the parameter to force quote gen to be in-proc or out-of-proc (override users' setting). Something like --proc <in|out>

Comment (qiucwang) [reply tests/tools/oecert/host/host.cpp:30]: Make sense.

File: tests/tools/oecert/host/host.cpp
Comment (yentsanglee) [reply tests/tools/oecert/host/host.cpp:438]: Miscommunicated. In the future given proper configuration, out-of-proc quote gen should work in Windows. Enforcing in-proc/out-of-proc should apply to both Linux and Windows. For now we only test out-of-proc quote gen in Linux.

File: tests/tools/oecert/host/host.cpp
Comment (yentsanglee) [inline tests/tools/oecert/host/host.cpp:433]: We should restore whatever is set/unset to SGX_AESM_ADDR. I.e.

char* sgx_aesm_env = getenv(SGX_AESM_ADDR);
...
setenv(SGX_AESM_ADDR, agx_aesm_env, 1);  // Restore value

Comment (qiucwang) [reply tests/tools/oecert/host/host.cpp]: Updated.

**Review by yentsanglee — APPROVED** [review]

**Comment by yentsanglee** [comment]
bors try

**Comment by qiucwang** [comment]
Jenkis host-verify out-of-proc quoting testing failed due to sgx-quote-ex lib not installed. Please merge this PR after #3882 merged.

**Comment by qiucwang** [comment]
bors try

**Comment by qiucwang** [comment]
bors try

**Review by mingweishih — APPROVED** [review]

**Review by ryanhsu19 — APPROVED** [review]
Thanks for the work!

**Pull request merged**
